cli/tests/unit/net_test.ts

```diff
@@ -807,3 +808,62 @@ Deno.test(
     assertEquals(res, "hello world!");
   },
 );
+
+Deno.test(
+  { permissions: { net: true } },
+  async function netListenUnref() {
+    const [statusCode, _output] = await execCode(`
+      async function main() {
+        const listener = Deno.listen({ port: 3500 });
+        listener.unref();
+        await listener.accept(); // This doesn't block the program from exiting
+      }
+      main();
+    `);
+    assertEquals(statusCode, 0);
+  },
+);
+
+Deno.test(
+  { permissions: { net: true } },
+  async function netListenUnref() {
+    const [statusCode, _output] = await execCode(`
+      async function main() {
+        const listener = Deno.listen({ port: 3500 });
+        await listener.accept();
+        listener.unref();
+        await listener.accept(); // The program exits here
```

Comment:
I believe this code is buggy - the second `listener.accept()` should result in an error due to resource being busy:
https://github.com/denoland/deno/blob/b4109375565c6d30d8ffb63f4083e0a430aee201/ext/net/ops.rs#L127-L135

Reply:
I think this 'accept' call is not performed concurrently with the previous one (the previous one is awaited, and it should be finished here).

Added a test case to ensure that concurrent 'accept' calls still throw 'Busy' while listener is unref'd. https://github.com/denoland/deno/pull/13961/commits/003f9bb4c45fe73f4e58ae3ee97084ad4ae29088